### PR TITLE
Solving merge conflicts

### PR DESCRIPTION
**Pull request by josenavas**
These are the files and functions that I needed to touch in order to solve the merge conflicts:
 - util.py
   * convert_to_id
   * convert_from_id
 - user.py
   * verify_code
   * create
   * default_analysis
   * user_studies
   * shared_studies
   * private_analyses
   * shared_analyses
 - environment_manager.py
   * make_environment
 - base.py
   * _check_id
   * __init__
   * _check_portal
 - study.py
   * get_by_status
   * get_info
   * create
   * delete
   * _portals
 - analysis.py
   * get_by_status
   * create
   * delete
   * exists
   * _portals
 - search.py
   * Merge conflict was only on imports

### CODE REVIEW

**Comment by josenavas** [comment]
BTW, any future PR in this branch is going to be based on this one.

File: qiita_db/portal.py [hunk 1]
```diff
@@ -0,0 +1,434 @@
+# -----------------------------------------------------------------------------
```
Comment (squirrelo) [inline qiita_db/portal.py]: Is the portals object supposed to be getting transaction-ed here or in a later pull request?

**Comment by squirrelo** [comment]
One question, :+1:  otherwise.

File: qiita_db/portal.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/portal.py]: Later PR

**Pull request merged**